x/ccv/provider/keeper/keeper.go

```diff
@@ -433,24 +433,26 @@ func (h StakingHooks) AfterUnbondingInitiated(ctx sdk.Context, ID uint64) {
 		consumerChainIDS = append(consumerChainIDS, chainID)
 		return false
 	})
-	valsetUpdateID := h.k.GetValidatorSetUpdateId(ctx)
-	unbondingOp := ccv.UnbondingOp{
-		Id:                      ID,
-		UnbondingConsumerChains: consumerChainIDS,
-	}
+	if len(consumerChainIDS) > 0 {
+		valsetUpdateID := h.k.GetValidatorSetUpdateId(ctx)
+		unbondingOp := ccv.UnbondingOp{
+			Id:                      ID,
+			UnbondingConsumerChains: consumerChainIDS,
+		}
 
-	// Add to indexes
-	for _, consumerChainID := range consumerChainIDS {
-		index, _ := h.k.GetUnbodingOpIndex(ctx, consumerChainID, valsetUpdateID)
-		index = append(index, ID)
-		h.k.SetUnbondingOpIndex(ctx, consumerChainID, valsetUpdateID, index)
-	}
+		// Add to indexes
+		for _, consumerChainID := range consumerChainIDS {
+			index, _ := h.k.GetUnbodingOpIndex(ctx, consumerChainID, valsetUpdateID)
+			index = append(index, ID)
+			h.k.SetUnbondingOpIndex(ctx, consumerChainID, valsetUpdateID, index)
+		}
 
-	// Set unbondingOp
-	h.k.SetUnbondingOp(ctx, unbondingOp)
+		// Set unbondingOp
+		h.k.SetUnbondingOp(ctx, unbondingOp)
 
-	// Call back into staking to tell it to stop this op from unbonding when the unbonding period is complete
-	h.k.stakingKeeper.PutUnbondingOnHold(ctx, ID)
+		// Call back into staking to tell it to stop this op from unbonding when the unbonding period is complete
+		h.k.stakingKeeper.PutUnbondingOnHold(ctx, ID)
```

Comment:
what happens theoretically if the unbonding is put on hold while child chains exist, but then child chain gets removed with gov proposal before hold is lifted?

Reply:
Nice catch @AdityaSripal